Automattic/simplenote-ios
# Update Tag List

### Details
This PR converts `SPTagsListViewController.m` to `TagListViewController.swift`
All the logic is left intact except for one moment:
* There is no longer a delay between callback from `FetchResultsController` and `tableView.reloadData()`. The delay was causing issues with data consistency as for a short (but significant) moment data in the table and data in the controller were different.
* As a result of the previous point animation of deleting a tag is gone. It will be back once we switch to ResultsController

@jleandroperez Please take a look! Thanks!

Closes #946 

Test scenarios found in previous pull requests:

### Testing: UI
1. Log into your account
2. Press over the top left button to reveal the Tags List

- [x] Verify that the `Settings` button shows up below `Trash`
- [x] Verify that Dark Mode looks as expected
- [x] Verify that Dynamic Type gets everything to properly resize
- [x] Verify that the Tags List properly remembers the previously selected cell (close and reopen)

### Testing: Tag Edition via Edition Button
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Press over the top left button to reveal the Tags List
5. Press the `Edit` button

- [x] Verify that **only** tag rows can be moved
- [x] Verify that **only** tag rows can be deleted
- [x] Verify that **only** tag rows can be renamed
- [x] Verify that while a Tag row is being renamed, the tableView's bottom insets are properly adjusted: you should be able to scroll down to the bottom.

### Testing: Tag Edition via MenuController
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Perform a long press gesture over any of the tags

- [x] Verify that the highlighted tag can be renamed
- [x] Verify that the highlighted tag can be deleted
- [x] Verify that a long press over `All Notes` / `Trash` / `Settings` do not trigger such menu

### Testing: Renaming
1. Make sure you have 3 or more tags
2. Open Sidebar
3. Rename first tag to an empty string
4. Instead of pressing `Done`, press on the second tag

- [x] Verify that the first tag is gone
- [x] Verify that the second tag is still in the list

### Testing: Tags Sort Order
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Press over the top left button to reveal the Tags List
5. Press over `Settings`
6. Change the `Tags` sort order

- [x] Verify that the Tags List gets properly updated

### Testing: Hiding Tags Sections
1. Log into a testing account with no Tag

- [x] Verify that an account with no Tags doesn't show the `Tags` section
- [x] Verify that an account with no Tags doesn't show the `Untagged` section

### Testing: Predicates!
1. Log into your account
2. Add three notes (with random contents)
3. Tag one of the new notes
4. Trash one of the new notes
5. And of course, leave one of them untagged

- [x] Verify the `Trash` filter shows the trashed note
- [x] Verify the `Untagged` filter shows the untagged note
- [x] Verify that tapping over the tag you've set in (3) properly displays the new note
